File: python/perfect-numbers/perfect_numbers.py
def classify(number: int) -> str:
    """A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <= 0:
        raise ValueError("Classification is only possible for positive integers.")

    factors = [num for num in range(1, number) if number % num == 0]
    aliquot_sum = sum(factors)
    if number == aliquot_sum:
        return "perfect"
    elif number < aliquot_sum:
        return "abundant"
    else:
        return "deficient"

File: go/perfect-numbers/perfect_numbers.go
package perfect

import "errors"

type Classification string

const (
	ClassificationPerfect   = "ClassificationPerfect"
	ClassificationAbundant  = "ClassificationAbundant"
	ClassificationDeficient = "ClassificationDeficient"
)

var ErrOnlyPositive = errors.New("Number cannot be negative")

func factors(n int64) []int64 {
	var factors []int64
	for i := int64(1); i < n; i++ {
		if n%i == 0 {
			factors = append(factors, i)
		}
	}
	return factors
}

func Classify(n int64) (Classification, error) {
	if n <= 0 {
		return "", ErrOnlyPositive
	}

	factors := factors(n)
	var sum int64
	for _, factor := range factors {
		sum += factor
	}

	if n == sum {
		return ClassificationPerfect, nil
	} else if n < sum {
		return ClassificationAbundant, nil
	} else {
		return ClassificationDeficient, nil
	}
}
